crates/bin/pcli/src/command/view/noble_address.rs

```diff
@@ -22,31 +36,243 @@ impl NobleAddressCmd {
+    // Noble Randomizer: [0xff; 10] followed by LE16(sequence)
+    let mut randomizer: [u8; 12] = [0xff; 12];
+    let seq_bytes = sequence.to_le_bytes();
+    randomizer[10..].copy_from_slice(&seq_bytes);
+
+    let index = AddressIndex {
+        account: account.unwrap_or_default(),
+        randomizer,
+    };
+
+    let (address, _dtk) = fvk.incoming().payment_address(index.into());
+
+    address
+}
+
+async fn register_noble_forwarding_account(
+    noble_node: &Url,
+    noble_address: &NobleForwardingAddress,
+    address: &Address,
+    channel: &str,
+) -> Result<NobleRegistrationResponse> {
+    let mut noble_client = CosmosServiceClient::new(
+        Channel::from_shared(noble_node.to_string())?
+            .tls_config(ClientTlsConfig::new())?
+            .connect()
+            .await?,
+    );
+
+    let tx = CosmosTx {
+        body: Some(CosmosTxBody {
+            messages: vec![pbjson_types::Any {
+                type_url: MsgRegisterAccount::type_url(),
+                value: MsgRegisterAccount {
+                    signer: noble_address.to_string(),
+                    recipient: address.to_string(),
+                    channel: channel.to_string(),
+                }
+                .encode_to_vec()
+                .into(),
+            }],
+            memo: "".to_string(),
+            timeout_height: 0,
+            extension_options: vec![],
+            non_critical_extension_options: vec![],
+        }),
+        auth_info: Some(CosmosAuthInfo {
+            signer_infos: vec![CosmosSignerInfo {
+                public_key: Some(pbjson_types::Any {
+                    type_url: ForwardingPubKey::type_url(),
+                    value: ForwardingPubKey {
+                        key: noble_address.bytes(),
+                    }
+                    .encode_to_vec()
+                    .into(),
+                }),
+                mode_info: Some(ModeInfo {
+                    // SIGN_MODE_DIRECT
+                    sum: Some(Sum::Single(Single { mode: 1 })),
+                }),
+                sequence: 0,
+            }],
+            fee: Some(CosmosFee {
+                amount: vec![],
+                gas_limit: 200000u64,
+                payer: "".to_string(),
+                granter: "".to_string(),
+            }),
+            tip: None,
+        }),
+        signatures: vec![vec![]],
+    };
+    let r = noble_client
+        .broadcast_tx(CosmosBroadcastTxRequest {
+            tx_bytes: tx.encode_to_vec().into(),
+            // sync
+            mode: 2,
+        })
+        .await?
+        .into_inner();
+
+    let code = r
+        .tx_response
+        .ok_or_else(|| anyhow::anyhow!("no tx response"))?
+        .code;
+
+    match code {
+        9 => Ok(NobleRegistrationResponse::NeedsDeposit),
+        0 => Ok(NobleRegistrationResponse::Success),
+        19 => Ok(NobleRegistrationResponse::AlreadyRegistered),
+        _ => Err(anyhow::anyhow!("unknown response from Noble")),
```

Comment:
On web, we don't get the `code 19`, but below when submitting a tx with the same sequence fyi. Looks like a [cometbft mempool err](https://github.com/cometbft/cometbft/blob/4eb7c058dd99c092b932c79d7cae33010f57cad9/mempool/errors.go#L12).

```
{"code":-32603,"message":"Internal error","data":"tx already exists in cache"}
```